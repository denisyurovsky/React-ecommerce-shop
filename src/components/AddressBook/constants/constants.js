export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
