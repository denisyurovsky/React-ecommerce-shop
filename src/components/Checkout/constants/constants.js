export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
