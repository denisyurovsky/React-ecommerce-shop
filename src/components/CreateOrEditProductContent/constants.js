export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
