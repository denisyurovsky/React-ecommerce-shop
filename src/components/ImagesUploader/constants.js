export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
