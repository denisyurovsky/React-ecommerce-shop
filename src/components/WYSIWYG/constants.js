export const TYPES = {
  LINK: 'LINK',
  CELL: 'CELL',
  EMAIL: 'EMAIL',
  TABLE: 'TABLE',
};

export const table = {
  MAX_ROWS: 20,
  COLUMNS: 2,
  HEADER: 'Product details',
  INPUT_ERROR: 'Max 20 rows',
};

export const MAX_LENGTH_MESSAGE = 'Max size is reached';
export const LINK_PREDEFINED_TEXT = 'https://';
export const EDITOR_MAX_SYMBOLS = 1000;
export const EDITOR_WARN_SYMBOLS = 50;

export const BREAKPOINTS = {
  FOOTER: 550,
  NAVIGATION: 350,
};
