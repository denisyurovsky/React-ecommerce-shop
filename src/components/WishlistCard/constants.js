export const DEFAULT_LABEL_INPUT = 'Wishlist Name';
export const ERROR_LABEL_INPUT = 'Wishlist name should be unique';
