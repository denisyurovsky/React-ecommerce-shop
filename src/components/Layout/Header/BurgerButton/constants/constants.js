import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
