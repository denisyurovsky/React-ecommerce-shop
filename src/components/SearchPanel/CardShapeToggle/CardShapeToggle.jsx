import ViewListIcon from '@mui/icons-material/ViewList';
import ViewModuleIcon from '@mui/icons-material/ViewModule';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import PropTypes from 'prop-types';
import React from 'react';

import { pageView } from '../../../pages/ProductListPage/constants/constants';

export default function CardShapeToggle({
  cardShape,
  setCardShape,
  className,
}) {
  const { LIST_VIEW, MODULE_VIEW } = pageView;

  const handleCardShape = (event, newCardShape) => {
    if (newCardShape !== null) {
      setCardShape(newCardShape);
    }
  };

  return (
    <ToggleButtonGroup
      className={className}
      value={cardShape}
      exclusive
      onChange={handleCardShape}
      aria-label="card shape"
    >
      <ToggleButton value={LIST_VIEW} aria-label={LIST_VIEW}>
        <ViewListIcon color="primary" />
      </ToggleButton>
      <ToggleButton value={MODULE_VIEW} aria-label={MODULE_VIEW}>
        <ViewModuleIcon color="primary" />
      </ToggleButton>
    </ToggleButtonGroup>
  );
}

CardShapeToggle.propTypes = {
  cardShape: PropTypes.string.isRequired,
  setCardShape: PropTypes.func.isRequired,
  className: PropTypes.string,
};
