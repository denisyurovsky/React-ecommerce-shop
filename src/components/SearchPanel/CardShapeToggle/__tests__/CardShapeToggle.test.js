import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import { pageView } from '../../../../pages/ProductListPage/constants/constants';
import CardShapeToggle from '../CardShapeToggle';

describe('CardShapeToggle component', () => {
  describe('should render a valid snapshots', () => {
    it('renders a valid snapshot', () => {
      const setCardShape = jest.fn();
      const { asFragment } = render(
        <CardShapeToggle
          cardShape={pageView.LIST_VIEW}
          setCardShape={setCardShape}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('main group of tests', () => {
    it('module toggle button works properly', () => {
      let cardShape = pageView.LIST_VIEW;
      const setCardShape = jest.fn(
        (newCardShape) => (cardShape = newCardShape)
      );

      render(
        <CardShapeToggle cardShape={cardShape} setCardShape={setCardShape} />
      );
      const moduleButton = screen.getByRole('button', { name: /module/i });

      userEvent.click(moduleButton);
      expect(cardShape).toEqual(pageView.MODULE_VIEW);
    });
    it('list toggle button works properly', () => {
      let cardShape = pageView.MODULE_VIEW;
      const setCardShape = jest.fn(
        (newCardShape) => (cardShape = newCardShape)
      );

      render(
        <CardShapeToggle cardShape={cardShape} setCardShape={setCardShape} />
      );
      const listButton = screen.getByRole('button', { name: /list/i });

      userEvent.click(listButton);
      expect(cardShape).toEqual(pageView.LIST_VIEW);
    });
  });
});
