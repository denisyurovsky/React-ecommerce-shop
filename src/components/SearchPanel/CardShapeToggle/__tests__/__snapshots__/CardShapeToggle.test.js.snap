// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardShapeToggle component should render a valid snapshots renders a valid snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="card shape"
    class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-label="list"
      aria-pressed="true"
      class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="list"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        data-testid="ViewListIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 14h4v-4H3v4zm0 5h4v-4H3v4zM3 9h4V5H3v4zm5 5h13v-4H8v4zm0 5h13v-4H8v4zM8 5v4h13V5H8z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="module"
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
      tabindex="0"
      type="button"
      value="module"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        data-testid="ViewModuleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.67 5v6.5H9.33V5h5.34zm1 6.5H21V5h-5.33v6.5zm-1 7.5v-6.5H9.33V19h5.34zm1-6.5V19H21v-6.5h-5.33zm-7.34 0H3V19h5.33v-6.5zm0-1V5H3v6.5h5.33z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
